### PR TITLE
Update insightstore interface

### PR DESCRIPTION
**Pull request by Hosshii**
**What this PR does / why we need it**:
- Add deployment store interface
- Separate insight store interface into three parts. 
- Delete unused files.

**Which issue(s) this PR fixes**:

ref #1142

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by khanhtc1202** [comment]
👍 
/lgtm

**Comment by nghialv** [comment]
Nice update. Keep up your great work.

/approve

File: pkg/insight/insightstore/deploymentstore.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+	"github.com/pipe-cd/pipecd/pkg/model"
+)
+
+var (
+	errUnimplemented = errors.New("unimplemented")
+)
+
+type DeploymentStore interface {
+	// GetDeployments returns slice of Deployment sorted by startedAt ASC.
+	GetDeployments(ctx context.Context, projectID string, from, to int64, minimumVersion *model.InsightDeploymentVersion) ([]*model.InsightDeployment, error)
```
Comment (knanao) [inline pkg/insight/insightstore/deploymentstore.go:30]: I think it better this interface name is `List` because it's not acquired by PK.

Comment (knanao) [reply pkg/insight/insightstore/deploymentstore.go:30]: Ops, It's too late lol

Comment (Hosshii) [reply pkg/insight/insightstore/deploymentstore.go:30]: I'll fix it on another PR.

**Pull request merged**